config/v1/types_feature.go

```diff
@@ -29,12 +29,40 @@ var (
 	// TechPreviewNoUpgrade turns on tech preview features that are not part of the normal supported platform. Turning
 	// this feature set on CANNOT BE UNDONE and PREVENTS UPGRADES.
 	TechPreviewNoUpgrade FeatureSet = "TechPreviewNoUpgrade"
+
+	// CustomNoUpgrade allows the enabling or disabling of any feature. Turning this feature set on IS NOT SUPPORTED, CANNOT BE UNDONE, and PREVENTS UPGRADES.
+	// Because of its nature, this setting cannot be validated.  If you have any typos or accidentally apply invalid combinations
+	// your cluster may fail in an unrecoverable way.
+	CustomNoUpgrade FeatureSet = "CustomNoUpgrade"
 )
 
 type FeatureGateSpec struct {
+	FeatureGateSelection `json:",inline"`
+}
+
+// +union
+type FeatureGateSelection struct {
 	// featureSet changes the list of features in the cluster.  The default is empty.  Be very careful adjusting this setting.
 	// Turning on or off features may cause irreversible changes in your cluster which cannot be undone.
+	// +unionDiscriminator
+	// +optional
 	FeatureSet FeatureSet `json:"featureSet,omitempty"`
+
+	// customFeatureGates allows the enabling or disabling of any feature. Turning this feature set on IS NOT SUPPORTED, CANNOT BE UNDONE, and PREVENTS UPGRADES.
```

Comment:
When you say can't be undone; if I put in feature Foo and the kubelet doesn't start are you saying we have no way to back out the Foo feature, to restore the cluster back to working? 

Will this apply to all TechPreview items?

Reply:
> When you say can't be undone; if I put in feature Foo and the kubelet doesn't start are you saying we have no way to back out the Foo feature, to restore the cluster back to working?

It means that you cannot go from `CustomNoUpgrade` back to default.  You can *never* upgrade this cluster.



> Will this apply to all TechPreview items?

It already does.